src/language-handlebars/printer-glimmer.js

```diff
@@ -289,8 +291,21 @@ function print(path, options, print) {
       ]);
     }
     case "MustacheCommentStatement": {
+      const rawNodeContent = textLines[n.loc.start.line - 1].slice(
+        n.loc.start.column,
+        n.loc.end.column
+      );
+      const isLeftWhiteSpaceSensitive = rawNodeContent.startsWith("{{~");
+      const isRightWhitespaceSensitive = rawNodeContent.endsWith("~}}");
```

Comment:
AST do not have this information?

Reply:
[Unfortunately not](https://prettier.io/playground/#N4Igxg9gdgLgprEAuExgD8CE6C+OQA0IEADjAJbQDOyoAhgE4MQDuACowjSnQDYt0AnjSIAjBnTABrODADKJSeSgBzZDAYBXOETgBbUXAAmR4wBk6qzXRVwAYhAZ66MCquQg6mmBEIgAFjB6vADq-uTwVIpgcHJcEeQAbhGCHmBUIiDKVHAMMGwSKs7IAGZ8OUQAVlQAHgBCEtKycnR6cGbKcKXlOiDVNXLKKrxwAIqaEPDdvBUgigw5DB7D5HptS0QkDMowIeRGMP7IABwADJvMOSESJB5bcIuJXUQAjhPwBaTcnlQAtFBwYzGPwMOBvciggo2YpIMozXo5PTkdRaBFDEbjSZdWE9IgwOiiPYHI5IABMeIk5F4QwAwhA1nQPA8AKx+TQ5AAqBO4cNmiW0AEkoKZYHIwNsyABBYVyGCCEbTHJ4IA)

Thankfully, splitting code by newlines (once) is a quick and easy way to read characters at the offset that `loc` attribute gives us.

Reply:
It seems not possible to fix [this](https://github.com/prettier/prettier/blob/6d7fdc7605025d076d1364df7807963b9bb34d8d/src/language-handlebars/parser-glimmer.js#L45) without `originalText`?

Reply:
There is a `locationToOffset` function, Maybe we can use it?

Reply:
I saw `locationToOffset` but didn't fully understand its intent and it seemed too complex. From what I can see, it would also lead to iterating over source twice (once for start offset and once for end offset).

Reply:
Acutally there is another one....

https://github.com/prettier/prettier/blob/master/src/utils/line-column-to-index.js

We should DRY them, maybe in future.